### PR TITLE
Improve Remote Gateway field description for IPSec VPN Phase 1. Implements #7095

### PR DESCRIPTION
**Pull request by vktg**
- [X] Redmine Issue: https://redmine.pfsense.org/issues/7095
- [X] Ready for review

### CODE REVIEW

File: src/usr/local/www/vpn_ipsec_phase1.php [hunk 1]
```diff
@@ -732,7 +732,8 @@ function build_eal_list() {
 		'*Remote Gateway',
 		'text',
 		$pconfig['remotegw']
-	))->setHelp('Enter the public IP address or host name of the remote gateway.');
+	))->setHelp('Enter the public IP address or host name of the remote gateway. ' .
+		    'Use 0.0.0.0 to allow connection from any IPv4 address.');
```
Comment (jim-p) [inline src/usr/local/www/vpn_ipsec_phase1.php:736]: Needs more explanation about the limitations of doing that: Can't initiate, can't use the peer IP address for the remote identifier, and may be limited to a single entry.

Though the only one entry may have been address by PR #4173 but I don't know if anyone tested that with `0.0.0.0`. Test to confirm that.

The identifier/PSK issue may have a workaround (define a PSK for `%any` on the PSK tab) but that's probably not something we should encourage.

Comment (vktg) [reply src/usr/local/www/vpn_ipsec_phase1.php]: > Needs more explanation about the limitations of doing that: Can't initiate, can't use the peer IP address for the remote identifier, and may be limited to a single entry.

Added

> Though the only one entry may have been address by PR #4173 but I don't know if anyone tested that with `0.0.0.0`. Test to confirm that.

Multiple entries with '0.0.0.0' with different IDs works fine

> The identifier/PSK issue may have a workaround (define a PSK for `%any` on the PSK tab) but that's probably not something we should encourage.

No issues were found after adding Mobile IPsec with PSK% any - other site-to-site '0.0.0.0' connections successfully connect

File: src/usr/local/www/vpn_ipsec_phase1.php [hunk 2]
```diff
@@ -732,7 +744,12 @@ function build_eal_list() {
 		'*Remote Gateway',
 		'text',
 		$pconfig['remotegw']
-	))->setHelp('Enter the public IP address or host name of the remote gateway.');
+	))->setHelp('Enter the public IP address or host name of the remote gateway.%1$s%2$s%3$s',
+	    '<div class="infoblock">',
+	    sprint_info_box(gettext('Use \'0.0.0.0\' to allow connection from any IPv4 address or \'::\' ' .
```
Comment (jim-p) [inline src/usr/local/www/vpn_ipsec_phase1.php:749]: `connection` should be plural: `connections`

File: src/usr/local/www/vpn_ipsec_phase1.php [hunk 3]
```diff
@@ -732,7 +744,12 @@ function build_eal_list() {
 		'*Remote Gateway',
 		'text',
 		$pconfig['remotegw']
-	))->setHelp('Enter the public IP address or host name of the remote gateway.');
+	))->setHelp('Enter the public IP address or host name of the remote gateway.%1$s%2$s%3$s',
+	    '<div class="infoblock">',
+	    sprint_info_box(gettext('Use \'0.0.0.0\' to allow connection from any IPv4 address or \'::\' ' . 
+	    'to allow connection from any IPv6 address.' . '<br/>' . 'It supports only Responder mode, ' .
```
Comment (jim-p) [inline src/usr/local/www/vpn_ipsec_phase1.php:750]: `connection` should be plural: `connections`

File: src/usr/local/www/vpn_ipsec_phase1.php [hunk 4]
```diff
@@ -732,7 +744,12 @@ function build_eal_list() {
 		'*Remote Gateway',
 		'text',
 		$pconfig['remotegw']
-	))->setHelp('Enter the public IP address or host name of the remote gateway.');
+	))->setHelp('Enter the public IP address or host name of the remote gateway.%1$s%2$s%3$s',
+	    '<div class="infoblock">',
+	    sprint_info_box(gettext('Use \'0.0.0.0\' to allow connection from any IPv4 address or \'::\' ' . 
+	    'to allow connection from any IPv6 address.' . '<br/>' . 'It supports only Responder mode, ' . 
+	    'can\'t use the peer IP address for the remote identifier, and limited to a single entry.'), 'info', false),
```
Comment (jim-p) [inline src/usr/local/www/vpn_ipsec_phase1.php:751]: Starting with `It supports only Responder mode,` this needs reworded as it's a bit awkward/incorrect grammar.

Perhaps: "Responder Only must be set, Peer IP Address cannot be used for Remote Identifier, and only a single Phase 1 entry may use this Remote Gateway value."

Comment (vktg) [reply src/usr/local/www/vpn_ipsec_phase1.php]: ok, all fixed
but you can use 0.0.0.0 for multiple PH1 (tested, see above)

Comment (jim-p) [reply src/usr/local/www/vpn_ipsec_phase1.php]: If it works OK, that can be removed then.

Comment (vktg) [reply src/usr/local/www/vpn_ipsec_phase1.php]: 'and limited to a single entry.' is already removed

**Pull request merged**